qemu-test

```diff
@@ -45,9 +45,6 @@ check_kernel() {
 
 check_kernel || exit 1
 
-# make sure the tests are built
-make TESTS= check > /dev/null
-
```

Comment:
Why can't we adapt this to meson as well?

Reply:
Tests are currently either build or not by default, depending on whether "-Dtests=true" was configured or not. Thus it is less likely to have sources built without having tests built.

One could maybe invoke meson/ninja from within the script, but then we also need to know where the build dir is which is already a bit hacky solved in `qemu-test-init`.

Reply:
Don't we need to know the build dir anyway to find the binaries?

Reply:
The intention was to avoid building in qemu (which is potentially slow when running without kvm). As we hardcode `BUILD_DIR` anyway, we could continue to build the test binaries in `qemu-init`.